### PR TITLE
[IND-498]: Add TradingRewardsHandler

### PR DESCRIPTION
**Pull request by Christopher-Li**
### Changelist
Add support for Trading Rewards Handler

### Test Plan
Unit tests

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: indexer/services/ender/src/scripts/handlers/dydx_trading_rewards_handler.sql [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    - event_index: The 'event_index' of the IndexerTendermintEvent.
+    - transaction_index: The transaction_index of the IndexerTendermintEvent after the conversion that takes into
+        account the block_event (https://github.com/dydxprotocol/v4-chain/blob/9ed26bd/indexer/services/ender/src/lib/helper.ts#L41)
+    - transaction_hash: The transaction hash corresponding to this event from the IndexerTendermintBlock 'tx_hashes'.
+  Returns: JSON object containing fields:
+    - tradingRewards: A list of the trading rewards in the trading-reward-model format (indexer/packages/postgres/src/models/trading-reward-model.ts)
+
+  (Note that no text should exist before the function declaration to ensure that exception line numbers are correct.)
+DECLARE
+    trading_rewards_array trading_rewards%ROWTYPE;
```
Comment (dydxwill) [inline indexer/services/ender/src/scripts/handlers/dydx_trading_rewards_handler.sql:19]: what's this for? remove?

File: indexer/services/ender/src/validators/trading-rewards-validator.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      this.validateTradingReward(reward, index);
+    });
+  }
+
+  private validateTradingReward(
+    reward: AddressTradingReward,
+    index: number,
+  ): void {
+    const denoms: bigint = bytesToBigInt(reward.denomAmount);
+    if (denoms === 0n) {
```
Comment (dydxwill) [inline indexer/services/ender/src/validators/trading-rewards-validator.ts:21]: Convention in codebase is to use BigInt(0)

File: indexer/services/ender/src/scripts/helpers/dydx_process_trading_reward_event.sql [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  Returns: trading_rewards row that was created as a result of the trading_reward jsonb event
+
+  (Note that no text should exist before the function declaration to ensure that exception line numbers are correct.)
+*/
+DECLARE
+    trading_reward_record trading_rewards%ROWTYPE;
+    wallet_record wallets%ROWTYPE;
+    amount_in_human_readable numeric;
+BEGIN
+    amount_in_human_readable = dydx_trim_scale(dydx_from_serializable_int(trading_reward->'denomAmount') * power(10, -18)::numeric);
```
Comment (dydxwill) [inline indexer/services/ender/src/scripts/helpers/dydx_process_trading_reward_event.sql:22]: where is -18 coming from? Should we make it a constant?

File: indexer/services/ender/__tests__/handlers/trading-rewards-handler.test.ts [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+    const kafkaMessage: KafkaMessage = createKafkaMessageFromTradingRewardsEvent({
+      tradingRewardsEvent,
+      transactionIndex: defaultTransactionIndex,
+      height: defaultHeight,
+      time: defaultTime,
+      txHash: defaultTxHash,
+    });
+
+    await onMessage(kafkaMessage);
+    const firstTradingRewardId: string = TradingRewardTable.uuid(
```
Comment (dydxwill) [inline indexer/services/ender/__tests__/handlers/trading-rewards-handler.test.ts:114]: consider condensing with Promise.all

File: indexer/services/ender/src/scripts/handlers/dydx_trading_rewards_handler.sql (hunk 1 above)
Comment (Christopher-Li) [reply indexer/services/ender/src/scripts/handlers/dydx_trading_rewards_handler.sql:19]: oops my bad, this was when I was testing

**Review by dydxwill — APPROVED** [review]

**Pull request merged**